### PR TITLE
MM-18900 - Remove search unused status flags

### PR DESCRIPTION
**Pull request by nadalfederer**
#### Summary

Remove search unused status flags - requests.search
#### Ticket Link

one part to close mattermost/mattermost-server#12522
that https://github.com/mattermost/mattermost-redux/pull/1026 depends on this PR

#### Device Information
This PR was tested on:

### CODE REVIEW

File: app/screens/search/search.js [hunk 1]
```diff
@@ -510,13 +515,22 @@ export default class Search extends PureComponent {
             per_page: 20,
             include_deleted_channels: viewArchivedChannels,
         };
-        actions.searchPostsWithParams(currentTeamId, params, true);
+        this.setState({isLoading: true, isLoaded: false, status: 'isLoading'});
+        const {error} = await actions.searchPostsWithParams(currentTeamId, params, true);
+
+        // actions.searchPostsWithParams(currentTeamId, params, true);
```
Comment (enahum) [inline app/screens/search/search.js]: missed ?

**Comment by nadalfederer** [comment]
>Snapshot tests should be updated to reflect snapshots for all use cases.

I'm sorry, I don't get what that means. I have updated snapshots. Are they wrong?

**Comment by enahum** [comment]
@nafalfederer they are not wrong, I’m just asking to add more tests to cover all cases to have a snapshot for them in case anything else changes in the future

**Comment by nadalfederer** [comment]
thanks, I'll add more tests

File: app/screens/flagged_posts/flagged_posts.test.js [hunk 2]
```diff
@@ -29,20 +29,58 @@ describe('FlaggedPosts', () => {
...
     });
 
+    test('should match snapshot when getFlaggedPosts failed', async () => {
+        const error = new Error('foo');
+
+        const newProps = {
+            ...baseProps,
+            actions: {
+                ...baseProps.actions,
+                getFlaggedPosts: jest.fn(async () => ({error})),
```
Comment (saturninoabril) [inline app/screens/flagged_posts/flagged_posts.test.js]: This works but I suggest to use jest convention.  Please also update other similar instances.
```suggestion
                getFlaggedPosts: jest.fn().mockResolvedValue({error}),
```

**Comment by enahum** [comment]
Hey @saturninoabril @migbot have we built and test this PR?

**Comment by saturninoabril** [comment]
@enahum Yes I did, particularly did tests around pin, flag, mention and search functionality.  Looks good to me.

**Pull request merged**